src/util/change-types.ts

```diff
@@ -30,7 +31,9 @@ export const changeStationType = (
 ) => {
     const currentStnType = graph.getNodeAttribute(selectedFirst, 'type') as StationType;
     const names = graph.getNodeAttribute(selectedFirst, currentStnType)!.names;
-    const newAttrs = { ...stations[newStnType].defaultAttrs, names };
+    const tmpNames = Object.values(StationsWithOneName).includes(currentStnType) ? [names[0], 'Stn'] : names;
+    const newNames = Object.values(StationsWithOneName).includes(newStnType) ? [tmpNames[0]] : tmpNames;
```

Comment:
How do we deal with names that has 3 items? We can use the default names and copy the old names one by one in a loop.